code/modules/admin/DB_ban/functions.dm

```diff
@@ -398,25 +398,30 @@
 	output += "<input type='hidden' name='src' value='[REF(src)]'>"
 	output += HrefTokenFormField()
 	output += "<b>Ckey:</b> <input type='text' name='dbsearchckey' value='[playerckey]'>"
-	output += "<b>Admin ckey:</b> <input type='text' name='dbsearchadmin' value='[adminckey]'>"
+	output += "<b>Admin ckey:</b> <input type='text' name='dbsearchadmin' value='[adminckey]'><br>"
+	output += "<b>IP:</b> <input type='text' name='dbsearchip' value='[ip]'>"
+	output += "<b>CID:</b> <input type='text' name='dbsearchcid' value='[cid]'>"
 	output += "<input type='submit' value='search'>"
 	output += "</form>"
 	output += "Please note that all jobban bans or unbans are in-effect the following round."
 
-	if(adminckey || playerckey)
+	if(adminckey || playerckey || ip || cid)
 		playerckey = sanitizeSQL(ckey(playerckey))
 		adminckey = sanitizeSQL(ckey(adminckey))
-		var/playersearch = ""
-		var/adminsearch = ""
+		var/search = ""
 		if(playerckey)
-			playersearch = "AND ckey = '[playerckey]' "
+			search += "AND ckey = '[playerckey]' "
 		if(adminckey)
-			adminsearch = "AND a_ckey = '[adminckey]' "
+			search += "AND a_ckey = '[adminckey]' "
+		if(ip)
+			search += "AND ip = '[ip]' "
+		if(cid)
+			search += "AND computerid = '[cid]' "

```

Comment:
sql injection exploit, ip and cid are not sanitized

Reply:
Also, a better way to do query building like this is to either build an associated key => value list and then build it at the end in a for loop, or just a list of where conditions and then just do `search.Join(" AND")`